### ISSUE TITLE
Echidna cannot find assertion failures outside of main contract

### ISSUE THREAD

**Issue by hacker-DOM** [crytic/echidna#668]
Consider this setup:

```
contract A {
    function foo() public {
        assert(false);
    }
}

contract C {
    constructor() public {
        A a = new A();
    }

    bool public echidna_test = true;
}
```

called with

```
echidna-test . --contract C --check-asserts --multi-abi
```

The test passes.

I believe this is because of the following:
- the `ts` variable that is passed to `ui` comes from `Solidity.prepapreForTest`
- tests is the last element in the return value here: https://github.com/crytic/echidna/blob/b11be00956341f487b10a580de90ef8fe46e8c41/lib/Echidna/Solidity.hs#L275-L276
- both the `ts` variable and the `a` variable there come from `Solidity.loadSpecified`
- they are the second and third last elements in the return value here: https://github.com/crytic/echidna/blob/b11be00956341f487b10a580de90ef8fe46e8c41/lib/Echidna/Solidity.hs#L239
- `neFuns` is public functions of the main contract, except those that start with `echidna_` and those that are filtered out, and the fallback function
- `tests` are public functions that start with `echidna_`
- it follows that the current behavior is:
- if `checkAsserts` is *not* enabled , then only `tests` of the main contract will be the tests
- if `checkAsserts` *is* enabled, then those `tests` and `neFuns` will be the tests
- tldr: echidna does not currently include functions outside of the main contract in its tests
- so far everything has been about *tests*
- from my experimentation, it seems that Echidna does not catch assertion failures in executing functions (before tests are run after every transaction)
- it follows the behavior is that Echidna does not catch the above "bug"

**Comment by ggrieco-tob** [comment]
Duplicated of #601.
